Makefile

```
@@ -1,3 +1,5 @@
+CXXFLAGS=-std=c++17
```

A well-behaved Makefile should respect `CXXFLAGS` user has set in the environment:
```suggestion
CXXFLAGS:=-std=c++17 $(CXXFLAGS)
```
Note that user's `CXXFLAGS` are added in the end where they will have the power to override `-std` if needed.

---

Shouldn't we also add [flags related to `nlohmann/json` headers](https://json.nlohmann.me/integration/pkg-config/)?